l10n_br_account/models/account_invoice.py

```diff
@@ -316,6 +316,94 @@ def onchange_fiscal_document_id(self):
         if self.issuer == '0':
             self.document_serie_id = self.company_id.document_serie_service_id
 
+    @api.multi
+    def invoice_validate(self):
+        super(AccountInvoice, self).invoice_validate()
+        for object in self.env['account.invoice'].search(
+                [('id', '=', self.id)]):
+            if (not object.journal_id.revenue_expense
+                and object.journal_id.automatic_conciliation
+                and object.fiscal_category_id.property_journal ==
+                    object.journal_id
+                and object.state == 'open'
+                and object.company_id.id ==
+                    self.env.user.company_id.id
+                and object.journal_id.conciliation_journal):
+                voucher_obj = self.env['account.voucher']
```

Comment:
@danimaribeiro 

A maneira realmente correta de fazer isto seria através de stock movements, não usando uma invoice para o efeito. Até porque não é nada mais do que um Transporte de Mercadoria, seja para industrialização ou para exposição...

Acredito que os movimentos invertidos seriam uma boa abordagem usando aquilo que temos no momento mas para tal.

Ainda assim, neste momento o Odoo permite adicionar uma conta de tipo recebível no account.invoice.account_id e de tipo resultado no account.invoice.line.account_id, fazendo um lançamento de débito na conta recebível e um crédito na conta resultado.

Quando efetuamos um pagamento (se a conta recebível tiver account.account.reconcile = 1), ele credita nessa mesma conta, e debita qualquer outra conta no account.journal.account.default_debit_account_id.

Então eu acho que se configurarmos um diário (como o diário banco) que no account.journal.account.default_debit_account_id tenha a mesma conta que no account.invoice.line.account_id, conseguimos fazer o movimento inverso anulando os movimentos contábeis. Configurando conciliação automática, torna-se fácil e rápido usar este processo sem qualquer alteração no código. Estou verificando com os nossos contadores.

Isto para uma solução prática e rápida, pois na verdade a maneira mais correta seria realmente apenas fazer movimentações de stock e poder emitir o Documento Fiscal para o seu transporte de lá mesmo, até para não forçar a criação de filtros desnecessários quando analisando os dados. A pensar para o futuro.